int8 y_speed
int8 x_speed
